### PR TITLE
Occurrences List

### PR DESCRIPTION
**Pull request by Lima21**
Fixes issue #25

### CODE REVIEW

File: lib/di/dependency_provider.dart [hunk 1]
```diff
@@ -110,28 +120,33 @@ class DependencyProvider extends InheritedWidget {
...
     _parishManager = ParishManager(parishService, parishListResponseMapper);
     _countyManager = CountyManager(countyService, countyListResponseMapper);
-    _districtManager = DistrictManager(districtService, districtListResponseMapper);
+    _districtManager =
+        DistrictManager(districtService, districtListResponseMapper);
     _statusManager = StatusManager(statusService, statusListResponseMapper);
     _typesManager = TypesManager(typesService, typeListResponseMapper);
     _speciesManager = SpeciesManager(speciesService, speciesListResponseMapper);
     _familyManager = FamilyManager(familyService, familyListResponseMapper);
-    _occurrencesManager = OccurrencesManager(occurrenceService, occurrenceListResponseMapper);
```
Comment (Vanethos) [inline lib/di/dependency_provider.dart:134]: When formatting files, please do it in another PR since it makes the life of the reviewers easier :P

File: lib/presentation/ui/occurrences/occurrences_list_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  /// Event to relay MockData information to the UI
+  var _mockDataSubject = BehaviorSubject<List<OccurrenceModel>>();
+
+  Stream<List<OccurrenceModel>> get mockDataStream => _mockDataSubject.stream;
+
+  OccurrencesListBloc(this._countyManager) {
+    disposable.add(_fetchNewDataSubject.stream
+        .flatMap((_) => _countyManager.getRecentOccurrences())
+        .map((base) => base.toList())
+        .listen(_mockDataSubject.add));
```
Comment (Vanethos) [inline lib/presentation/ui/occurrences/occurrences_list_bloc.dart]: There  needs to be an error handler here, else when the API fetch throws an error, who will handle the error?

You can use some of the BaseBloc's methods to relay the message, such as `_errorSubject`

File: lib/presentation/ui/occurrences/occurrences_list_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+class OccurrencesListBloc extends BaseBloc {
+  OccurrencesManager _countyManager;
+
+  /// Event to fetch new data
+  var _fetchNewDataSubject = PublishSubject<Event>();
+
+  Sink<Event> get fetchNewDataSink => _fetchNewDataSubject.sink;
+
+  /// Event to relay MockData information to the UI
+  var _mockDataSubject = BehaviorSubject<List<OccurrenceModel>>();
```
Comment (Vanethos) [inline lib/presentation/ui/occurrences/occurrences_list_bloc.dart]: Since we are not dealing with mock data anymore, please give it an appropriate name to reflect what it is handling

File: lib/presentation/ui/occurrences/occurrences_item.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:vost/domain/models/occurrence_model.dart';
+import 'package:vost/presentation/assets/text_styles.dart';
+
+class OccurrencesItem extends StatefulWidget {
```
Comment (Vanethos) [inline lib/presentation/ui/occurrences/occurrences_item.dart]: Since you don't have any state being changed in this widget, can it be `Stateless`?

File: lib/presentation/ui/home/home_page.dart [hunk 5]
```diff
@@ -177,9 +182,16 @@ class _MyHomePageState extends BaseState<HomePage> {
   /// Callback to navigate to Report a Problem screen
   void _onReportTap() {
     //todo: navigate to report a problem
-    
+
     Navigator.of(context).pushNamed(routeProblem);
   }
+
+  /// Callback to navigate to Report a Problem screen
```
Comment (Vanethos) [inline lib/presentation/ui/home/home_page.dart]: The method name is for occurrences but the comments talk about Report a Problem :P

File: lib/presentation/ui/occurrences/occurrences_list_page.dart [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:vost/common/event.dart';
+import 'package:vost/domain/models/occurrence_model.dart';
+import 'package:vost/presentation/ui/_base/base_page.dart';
+import 'package:vost/presentation/ui/occurrences/occurrences_item.dart';
+import 'package:vost/presentation/ui/occurrences/occurrences_list_bloc.dart';
+
+class OccurrencesList extends BasePage<OccurrencesListBloc> {
```
Comment (Vanethos) [inline lib/presentation/ui/occurrences/occurrences_list_page.dart]: The occurrences list should be displayed in the home page widget, so is it possible to move it there?

At the same time, do do you think it's best to put it all in the homepage bloc since it is all present in the same context?

File: lib/presentation/ui/occurrences/occurrences_list_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import 'package:rxdart/rxdart.dart';
+import 'package:vost/common/event.dart';
+import 'package:vost/domain/managers/occurrences_manager.dart';
+import 'package:vost/domain/models/occurrence_model.dart';
+import 'package:vost/presentation/ui/_base/base_bloc.dart';
+
+class OccurrencesListBloc extends BaseBloc {
```
Comment (Vanethos) [inline lib/presentation/ui/occurrences/occurrences_list_bloc.dart]: See comments on the occurrences_list_page widget for discussion in how we should treat this bloc

**Review by Vanethos — CHANGES_REQUESTED** [review]

**Comment by Vanethos** [comment]
This screen will be reused in other parts of the app, such as the favorites list and the search.

However, since we can reuse the ItemWidget, maybe we can go without the OccurrencesListPage

Again, I'm sorry if I was not clear on this issue! It is my fault for not being more straight before

**Comment by Lima21** [comment]
> This screen will be reused in other parts of the app, such as the favorites list and the search.
> 
> However, since we can reuse the ItemWidget, maybe we can go without the OccurrencesListPage
> 
> Again, I'm sorry if I was not clear on this issue! It is my fault for not being more straight before

So, I've to update the PR only with the "occurrences_item.dart"?

File: lib/presentation/ui/occurrences/occurrences_item.dart (hunk 4 above)
Comment (Lima21) [reply lib/presentation/ui/occurrences/occurrences_item.dart]: I created as a StatefulWidget because we need to set the favorite, I don't know if it's the good place to do that, so I need to change to Stateless?

**Review by Vanethos — APPROVED** [review]

**Comment by Vanethos** [comment]
As we say in Portuguese "espectáculo!"

Approved 🥇
